### PR TITLE
GIX-1795: Add Dissolve Date in Advanced Section

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Add the dissolve date for dissolving neurons in the Advanced Section.

# Changes

* New utils to get the dissolve date from SNS and NNS neurons.
* New utils are used in another related util.
* New utils are used to get the dissolve date (if any) and render it in the SNS and SNS NeuronAdvancedSection.

# Tests

* Test new utils.
* Test new functionality in SNS and SNS NeuronAdvancedSection.

# Todos

Not yet necessary, it's covered by the entry of the new neuron details UI.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UI:

Screenshot 2023-08-09 at 10 30 38

**Comment by lmuntaner** [comment]
@dskloetd @peterpeterparker please review

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts [hunk 1]
```diff
@@ -22,11 +25,12 @@ import { mockRewardEvent } from "$tests/mocks/nns-reward-event.mock";
 import { NnsNeuronAdvancedSectionPo } from "$tests/page-objects/NnsNeuronAdvancedSection.page-object";
 import { JestPageObjectElement } from "$tests/page-objects/jest.page-object";
 import { normalizeWhitespace } from "$tests/utils/utils.test-utils";
-import type { NeuronInfo } from "@dfinity/nns";
+import { NeuronState, type NeuronInfo } from "@dfinity/nns";
 import { render } from "@testing-library/svelte";
 import NeuronContextActionsTest from "./NeuronContextActionsTest.svelte";
 
 describe("NnsNeuronAdvancedSection", () => {
+  const nowInSeconds = 1689843195;
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts:33]: Does it work to set
```
const nowInSeconds = new Date("Jul 20, 2023 8:53 AM").getTime() / 1000;
```
? Then it's easy to see that the expected date is correct.

File: frontend/src/tests/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.spec.ts [hunk 2]
```diff
@@ -20,6 +24,7 @@ import { mockToken } from "$tests/mocks/sns-projects.mock";
 import { SnsNeuronAdvancedSectionPo } from "$tests/page-objects/SnsNeuronAdvancedSection.page-object";
 import { JestPageObjectElement } from "$tests/page-objects/jest.page-object";
 import { normalizeWhitespace } from "$tests/utils/utils.test-utils";
+import { NeuronState } from "@dfinity/nns";
 import type { Principal } from "@dfinity/principal";
 import { SnsNeuronPermissionType, type SnsNeuron } from "@dfinity/sns";
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.spec.ts:30]: same for `nowInSeconds` below

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts:33]: I guess, no strong opinion.

Comment (lmuntaner) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts]: done

File: frontend/src/tests/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.spec.ts:30]: done

**Comment by lmuntaner** [comment]
@dskloetd please review

**Review by dskloetd — APPROVED** [review]

**Pull request merged**